src/encoded/types/workflow.py

```
@@ -942,7 +943,14 @@ def pseudo_run(context, request):
                             Payload=json.dumps(input_json))
     res_decode = res['Payload'].read().decode()
     res_dict = json.loads(res_decode)
-    arn = res_dict['_tibanna']['response']['executionArn']
+
+    # known to fail and propagate exception
+    try:
+        arn = res_dict['_tibanna']['response']['executionArn']
+    except Exception as e:
+        # XXX: do something
```

Is there not an obvious thing to do? What would help decide?

---

I think we should just throw the exception. Or should we return a response with the error message?

---

I'm going to raise `HTTPBadRequest` when this happens and report the response from tibanna and the exception.